client/app/components/navbar/navbar-notification.service.js

```diff
@@ -97,6 +98,7 @@ class NavbarNotificationService {
         name: 'notifications',
         title: this.$translate.instant('common_navbar_notification_title'),
         iconClass: 'icon-notifications',
+        iconAnimated: this.constructor.shouldAnimateIcon(sublinks),
```

Comment:
I wonder what is better: using `this.contructor.shouldAnimateIcon()` or `NavbarNotificationService.shouldAnimateIcon()`?

Reply:
Both are ok; `ClassName.method()` will always refer to `ClassName#method` while `this.constructor.method()` will refer to the class of the current instance (in case of inheritance).

```javascript
class A {
	static hello(){
		return 'Hello from A';
    }

    log(){
        console.log(A.hello());
    } 

    logConstructor(){
        console.log(this.constructor.hello());
    } 
    
} 

class B extends A {
    static hello(){
        return 'Hello from B';
    }
}

const a = new A();
a.log(); // Hello from A
a.logConstructor(); // Hello from A

const b = new B();
b.log(); // Hello from A
b.logConstructor(); // Hello from B
```